src/InputFieldMarkdownRenderChild.ts

```diff
@@ -34,38 +34,49 @@ export class InputFieldMarkdownRenderChild extends MarkdownRenderChild {
 	metadataValueUpdateQueue: any[];
 	inputFieldValueUpdateQueue: any[];
 
-	constructor(containerEl: HTMLElement, type: InputFieldMarkdownRenderChildType, fullDeclaration: string, plugin: MetaBindPlugin, filePath: string, uid: number) {
+	constructor(
+		containerEl: HTMLElement,
+		type: InputFieldMarkdownRenderChildType,
+		declaration: InputFieldDeclaration,
+		plugin: MetaBindPlugin,
+		filePath: string,
+		error?: string
+	) {
 		super(containerEl);
 
-		this.error = '';
+		this.error = error || '';
 		this.filePath = filePath;
-		this.uid = uid;
 		this.plugin = plugin;
 		this.type = type;
-		this.fullDeclaration = fullDeclaration;
+		this.fullDeclaration = declaration.fullDeclaration;
 
 		this.metadataValueUpdateQueue = [];
 		this.inputFieldValueUpdateQueue = [];
 		this.intervalCounter = 0;
-
-		try {
-			this.inputFieldDeclaration = InputFieldDeclarationParser.parse(fullDeclaration);
-
-			if (this.inputFieldDeclaration.isBound) {
-				this.parseBindTarget();
-				this.metaData = getFrontmatterOfTFile(this.bindTargetFile as TFile, this.plugin);
+		this.inputFieldDeclaration = declaration;
+		
+		if (isTruthy(error)) {
+			console.warn(error);
+		} else {
+			this.uid = this.plugin.markDownInputFieldIndex++;
```

Comment:
i am personally not a fan of the post-increment operator. It would be nice if you could refactor this to not use the post-increment operator

Reply:
absolutely!